### PR TITLE
Adding retry logic to kustomize apply

### PR DESCRIPTION
**Pull request by ben851**
## What happens when your PR merges?
The merge to staging github action will now use a shell script that contains retry logic when doing kubectl apply -k. This is required for karpenter because kustomize does not have a mechanism to wait for CRD creation or services entering ready state before proceeding to the next resource. 

This script will attempt to apply Kustomize. If it works, it will break out of a loop. If it fails, it will wait 10 seconds and retry. It will retry up to 5 times before erroring out fully.

## What are you changing?
- [X] Changing kubernetes configuration

## Provide some background on the changes
In support of K8s performance tuning and karpenter implementation

## Checklist if making changes to Kubernetes:
- [X] I know how to get kubectl credentials in case it catches on fire

### CODE REVIEW

File: scripts/applyKube.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+retryCount=0
+workDir=$1
+kubeConfig=$2
+pushd $workDir
```
Comment (sastels) [inline scripts/applyKube.sh:4]: nice

**Review by sastels — APPROVED** [review]
LGTM

File: scripts/applyKube.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+until [ "$retryCount" -ge 5 ]
+do
+   kubectl apply -k . $kubeConfig  && break
```
Comment (jimleroyer) [inline scripts/applyKube.sh:7]: Interesting way to break out of the loop, my shell programming is way far behind but that looks quite idiomatic to the language.

**Review by jimleroyer — APPROVED** [review]
Approved with a wish for some documentation on the new script.

**Pull request merged**